sets.go

```diff
@@ -27,24 +27,18 @@ func NewSet[T comparable](hasher Hasher[T], values ...T) Set[T] {
 //
 // This function will return a new set even if the set already contains the value.
 func (s Set[T]) Set(values ...T) Set[T] {
-	n := Set[T]{
-		m: s.m.clone(),
-	}
```

Comment:
How about ```n := NewSet(s.m.hasher, s.Items()...)```?

Reply:
That optimises for the case where the number of items you want to add is similar to the number of items already in the set.
I don't think that's necessarily a good idea.
As a user of the library I expect that adding $N$ items to a set takes time proportional to $N$, not to the original size of set.

Reply:
If you (as a user) _know_ that copying the current set and adding values with `mutable=true` will be faster than adding the values one by one, it's quite easy to implement this yourself in user code.

Reply:
fair